app/assets/stylesheets/components/_form.scss

```diff
@@ -76,6 +75,12 @@ input::-webkit-inner-spin-button {
   }
 }
 
+.error-state {
```

Comment:
I see we're now adding `usa-input--error` in JavaScript, so do we still need this?

Reply:
yea unfortunately until we remove the .fields basscss styling, we have an issue with the border colors being overwritten to the default border colors when it is invalid.

Reply:
I can possibly just add the things im doing here to the .usa-input--error class instead of creating a new class.

Reply:
Ah, I see. Maybe we could temporarily create a duplicate version of the [USWDS error styles](https://github.com/uswds/uswds/blob/08733d58255cb7cb98b61575d3697820fa08ce54/src/stylesheets/elements/form-controls/_text-input.scss#L11-L13) with a higher specificity to override BassCSS?

Something like:

```scss
.usa-input--error.field:invalid {
  @include u-border($theme-input-state-border-width, "error-dark");
}
```

Reply:
ah so I think its because of the value type :invalid. so until the input is invalid the border will be red since the input is invalid. Thats why I moved it over to trigger off aria-invalid.

Reply:
But with your new JavaScript, we're also adding `usa-input--error`, so the red border styles are being applied, they're just overridden by the BassCSS `:invalid` styling. If we use something like the snippet in my previous comment, we can make sure that the USWDS error styles take priority over the BassCSS invalid styles, without needing to add a new class `error-state`.